t3/simulate/rmg_simulator.py

```diff
@@ -0,0 +1,273 @@
+                 observable_list: Optional[list] = list(),
+                 ):
+
+        self.t3 = t3
+        self.rmg = rmg
+        self.paths = paths
+
+        self.logger = logger
+        # todo I don't think we need atol and rtol since these values are already presentin the rmg input.py file
+        # self.atol = self.rmg['model']['atol']
+        # self.rtol = self.rmg['model']['rtol']
+        self.observable_list = observable_list
+        self.sa_atol = self.t3['sensitivity']['atol']
+        self.sa_rtol = self.t3['sensitivity']['rtol']
+
+        # initialize other attributes
+        self.observable_species = list()
+        self.rmg_model = None
+        self.rmg_input_file = None
+
+        self.set_up()
+
+    def set_up(self):
+        """
+        Read in the chemkin file, species dictionary, and RMG input file.
+
+        If the user requested SA, simulate the job with SA, generate the output csv files,
+        and raise the following errors:
+            - ValueError: If RMG SA is not implemented for the given reactor type.
+
+        If SA is not requested, only simulate the job to obtain species profiles. Output is written to
+        a `solver` directory inside `rmg_analysis`.
+
+        """
+
+        # set up directories
+        if not os.path.isdir(self.paths['analysis_directory']):
+            os.mkdir(self.paths['analysis_directory'])
+        if len(self.observable_list):
+            # store SA results in an SA directory
+            self.rmg_input_file = self.paths['SA input']
+            if not os.path.isdir(self.paths['SA']):
+                os.mkdir(self.paths['SA'])
+            if not os.path.isdir(self.paths['SA solver']):
+                os.mkdir(self.paths['SA solver'])
+            if os.path.isfile(self.paths['SA input']):
+                os.remove(self.paths['SA input'])
+        else:
+            # store regular simulation results in solver directory that RMG creates automatically
+            self.rmg_input_file = os.path.join(self.paths['analysis_directory'], 'input.py')
+
+        # must copy the input file since load_rmg_py_job creates many directories based on the input file's location
+        if not os.path.isfile(self.rmg_input_file):
+            shutil.copyfile(src=self.paths['RMG input'], dst=self.rmg_input_file)
+
+        # create rmg object that all methods can access
+        self.rmg_model = load_rmg_py_job(input_file=self.rmg_input_file,
+                                         chemkin_file=self.paths['chem annotated'],
+                                         species_dict=self.paths['species dict'],
+                                         generate_images=True,
+                                         use_chemkin_names=False,
+                                         check_duplicates=False,
+                                         )
+        if self.rmg_model is None:
+            self.logger.error('The RMG adapter did not properly read the rmg input file.')
```

Comment:
shall we also return from this call (e.g., we'll crash on `for reaction_system in self.rmg_model.reaction_systems`)? and check this is not None in other method calls?